streaming-connect-common/src/test/java/com/adobe/platform/streaming/auth/impl/OAuth2IMSTokenProviderTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+package com.adobe.platform.streaming.auth.impl;
+
+import com.adobe.platform.streaming.auth.AuthException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * @author Adobe Inc.
+ */
+class OAuth2IMSTokenProviderTest {
```

Comment:
Add integration test for OAuth2 client credentials. Please refer AEPSinkConnectorTest.aepSinkConnectorJWTAuthenticationTest.